### PR TITLE
Migrate `claim` tests to hardhat

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: test/byContract/IexecPoco/06_claim.test.ts [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+            .to.emit(iexecPoco, 'Transfer')
+            .withArgs(iexecPoco.address, requester.address, taskPrice)
+            .to.emit(iexecPoco, 'Unlock')
+            .withArgs(requester.address, taskPrice)
+            .to.emit(iexecPoco, 'Seize')
+            .withArgs(scheduler.address, schedulerTaskStake, taskId)
+            .to.emit(iexecPoco, 'Reward')
+            .withArgs(kittyAddress, schedulerTaskStake, taskId)
+            .to.emit(iexecPoco, 'Transfer')
+            .withArgs(iexecPoco.address, kittyAddress, schedulerTaskStake)
```
Comment (zguesmi) [inline test/byContract/IexecPoco/06_claim.test.ts:187]:
```suggestion
            .to.emit(iexecPoco, 'Transfer')
            .withArgs(iexecPoco.address, kittyAddress, schedulerTaskStake)
            .to.emit(iexecPoco, 'Reward')
            .withArgs(kittyAddress, schedulerTaskStake, taskId)
```
To keep a consistent order.

File: test/byContract/IexecPoco/06_claim.test.ts [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+            .to.emit(iexecPoco, 'Lock')
+            .withArgs(kittyAddress, schedulerTaskStake);
```
Comment (zguesmi) [inline test/byContract/IexecPoco/06_claim.test.ts:189]:
```suggestion
            .to.emit(iexecPoco, 'Transfer')
            .withArgs(kittyAddress, iexecPoco.address, schedulerTaskStake)
            .to.emit(iexecPoco, 'Lock')
            .withArgs(kittyAddress, schedulerTaskStake);
```

File: test/byContract/IexecPoco/06_claim.test.ts [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        // No time traveling after deadline
+
+        await expect(iexecPoco.connect(anyone).claim(taskId)).to.be.reverted;
+    });
+
+    /**
+     * @notice Before properly matching orders, this method takes care of
+     * signing orders and depositing required stakes.
+     */
+    async function matchOrders(orders: IexecOrders) {
```
Comment (zguesmi) [inline test/byContract/IexecPoco/06_claim.test.ts:310]:
```suggestion
    async function signAndMatchOrders(orders: IexecOrders) {
```

Comment (jeremyjams) [reply test/byContract/IexecPoco/06_claim.test.ts:310]: I hesitated, good thanks :smiley:

File: test/byContract/IexecPoco/06_claim.test.ts (hunk 2 above)
Comment (jeremyjams) [reply test/byContract/IexecPoco/06_claim.test.ts:193]: Good catch on both thanks! See [Add/update event checks, factorize duration & rename match method](https://github.com/iExecBlockchainComputing/PoCo/pull/65/commits/0acfd9ef6c3219b7d3f49cb688f6359c66f636ba)

**Review by zguesmi — APPROVED** [review]
:heavy_check_mark:

**Review by gfournieriExec — APPROVED** [review]
Thanks Jérémy ! ;)

**Pull request merged**